pkg/vm/vm.go

```diff
@@ -1044,17 +1045,21 @@ func (v *VM) execute(ctx *Context, op opcode.Opcode, parameter []byte) (err erro
 			if base.Cmp(bigZero) <= 0 {
 				panic("invalid base")
 			}
-			if modulus.Cmp(bigTwo) < 0 {
+			if absModulus.Cmp(bigTwo) < 0 {
 				panic("invalid modulus")
 			}
-			if res.ModInverse(base, modulus) == nil {
+			if res.ModInverse(base, absModulus) == nil {
 				panic("base and modulus are not relatively prime")
 			}
 		case 1:
 			if modulus.Sign() == 0 {
 				panic("zero modulus") // https://docs.microsoft.com/en-us/dotnet/api/system.numerics.biginteger.modpow?view=net-6.0#exceptions
 			}
-			res.Exp(base, exponent, modulus)
+			res.Exp(base, exponent, absModulus)
```

Comment:
Why do you need an absolute modulus here? Isn't it enough to subtract `absModulus` from the result if the base is negative and exponent is even?

Reply:
I see that you removed this code, but could you explain why did you add this code in the first version of this PR?

Reply:
For the initial point, I added absModulus in res.Exp(base, exponent, absModulus) to explicitly ensure that the modulus used in modular exponentiation was positive. Also, I was looking at https://github.com/dotnet/runtime/blob/5535e31a712343a63f5d7d796cd874e563e5ac14/src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs#L968C13-L1023C27 and they have abs on trivialModulus.  However, it's redundant in Go implementation.